apps/web/src/app/[locale]/challenge/_components/challenge-layout.tsx

```diff
@@ -33,30 +32,70 @@ export const useLayoutSettingsStore = create<State>()(
 export interface ChallengeLayoutProps {
   left: ReactNode;
   right: ReactNode;
+  setIsDesktop: (bool: boolean) => void;
+  isDesktop: boolean;
+  leftSide: MutableRefObject<HTMLDivElement | null>;
+  collapsePanel: () => void;
+  expandPanel: () => void;
+  adjustPanelSize: (divideByW: number, divideByH: number, newDimensionValue: number) => void;
+  isLeftPanelCollapsed: () => boolean;
 }
 
-const MOBILE_BREAKPOINT = 1025;
-
-export function ChallengeLayout({ left, right }: ChallengeLayoutProps) {
+export const MOBILE_BREAKPOINT = 1025;
+
+export function ChallengeLayout({
+  left,
+  right,
+  setIsDesktop,
+  isDesktop,
+  leftSide,
+  adjustPanelSize,
+  collapsePanel,
+  expandPanel,
+  isLeftPanelCollapsed,
+}: ChallengeLayoutProps) {
   const parent = useRef<HTMLDivElement>(null);
   const resizer = useRef<HTMLDivElement>(null);
   const rightSide = useRef<HTMLDivElement>(null);
 
   const { settings, updateSettings } = useLayoutSettingsStore();
   const { fssettings, updateFSSettings } = useFullscreenSettingsStore();
-  const [isDesktop, setIsDesktop] = useState(window.innerWidth > MOBILE_BREAKPOINT);
   const [isCollapsed, setIsCollapsed] = useState(false);
 
   const LEFT_PANEL_BREAKPOINT = isDesktop ? 500 : 318;
   const COLLAPSE_BREAKPOINT = isCollapsed ? 50 : 300;
   const DEFAULT_DESKTOP_WIDTH_PX = `${LEFT_PANEL_BREAKPOINT}px`;
+  const COLLAPSED_DESKTOP_WIDTH = 60;
+  const COLLAPSED_MOBILE_HEIGHT = 41;
+  const leftStyleIfDesktopCollapsed = {
+    width: `${COLLAPSED_DESKTOP_WIDTH}px`,
+    minWidth: `${COLLAPSED_DESKTOP_WIDTH}px`,
+  };
+  const leftStyleIfMobileCollapsed = {
+    height: `${COLLAPSED_MOBILE_HEIGHT}px`,
+    minHeight: `${COLLAPSED_MOBILE_HEIGHT}px`,
+  };
+
+  const isPanelCollapsed = () => {
```

Comment:
Great idea, what do you think about something like this?

```typescript
  const isPanelCollapsed = useMemo(() => {
    const height = parseFloat(settings.height);
    const width = parseFloat(settings.width);

    return height <= COLLAPSED_MOBILE_HEIGHT || width <= COLLAPSED_DESKTOP_WIDTH;
  }, [settings]);

  const leftStyle = useMemo(() => {
    if (isDesktop) {
      return isPanelCollapsed
        ? leftStyleIfDesktopCollapsed
        : {
            width: settings.width,
            minWidth: `${LEFT_PANEL_BREAKPOINT}px`,
          };
    } else {
      return isPanelCollapsed
        ? leftStyleIfMobileCollapsed
        : {
            height: settings.height,
            minHeight: `${LEFT_PANEL_BREAKPOINT}px`,
          };
    }
  }, [[isPanelCollapsed, isDesktop, settings]]);
```

Let me know if this is too much / unreadable.

Reply:
you can probably nest the `isPanelCollapsed` inside the `leftStyle` right? i dont think you use it anywhere else but i may be mistaken

if you decide to keep it the same I think you want to use the primitives on the dep array `[settings.height, settings,width]`

Reply:
Yep, you're right and I think it looks cleaner! I just wanted to separate them to make it clear they're two different functions. I'll also update to use primitives.

For the dep array, is there anything in particular that is bad about using entire object? Is it slower for comparisons? Also, is it better in general to do this in case settings object grows?

Reply:
great question, heres an example

https://codesandbox.io/s/green-feather-38p78q?file=/src/App.js

notice how the bad continues to recalculate on button click. 
this is because if ANY property on the object is changed it will recalculate. 
if you explicitly define the properties you care about that are primitives it will only recalc if those change.

im unsure if theres additional properties on your object but even if there arent this will future proof in the event that additional props are added and changed

Reply:
Oki! I implemented it and everything works fine.

I just played a round a little more with it and I think I realized that this will always trigger no matter what, making the useMemo less useful. 
For example:
* Resizing the window in any way causes settings.width/settings.height to change
* Collapsing/expanding changes the settings.width/settings.height

I can't think of any situation where the dependencies will stay the same to memoize the value.

What do you think? Should we still use useMemo?

Reply:
it would only be helpful in the event that there are other state changes not caused by resize. if those dont exist then it can be removed probably